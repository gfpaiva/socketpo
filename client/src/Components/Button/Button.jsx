import React from 'react';
import PropTypes from 'prop-types';

import './Button.scss';

const Button = ({ children, className, medium, big, spaced, secondary, mx, ...props }) => (
	<button
		className={
			`${className ? `${className} ` : ''}
			button
			${big ? ' button--big' : ''}
			${medium ? ' button--medium' : ''}
			${spaced ? ' button--spaced' : ''}
			${secondary ? ' button--secondary' : ''}
			${mx ? ' button--mx' : ''}`}
		{...props}
	>
		{children}
	</button>
);

Button.propTypes = {
	className: PropTypes.string,
	medium: PropTypes.bool,
	big: PropTypes.bool,
	spaced: PropTypes.bool,
	secondary: PropTypes.bool,
	mx: PropTypes.bool
};

export default Button;

