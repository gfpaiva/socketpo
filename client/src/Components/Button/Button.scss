@import '../../Scss/root';

.button {
	background-color: $primary-color;
	border: none;
	box-shadow: 0px 0px 0px #000;
	color: $secondary-color;
	margin: 0;
	padding: 0;
	transition: box-shadow .3s ease;

	&--medium { @include responsive-font-size(22px) }
	&--spaced { padding: .25em 1em }
	&--mx { margin: 0 1em }
	&:hover { box-shadow: 0px 0px 30px #0000008c }
}
