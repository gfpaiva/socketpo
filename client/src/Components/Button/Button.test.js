import React from 'react';
import { mount } from 'enzyme';
import Button from './Button';

describe('<Button />', () => {
	it('should mount properly', () => {
		const wrapper = mount(<Button />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should mount with children properly', () => {
		const wrapper = mount(<Button>Test</Button>);
		expect(wrapper).toMatchSnapshot();
	});

	it('should mount with props "className, medium, big, spaced, secondary, mx, onClick"', () => {
		const wrapper = mount(
			<Button
				className="test"
				medium
				big
				spaced
				secondary
				mx
				onClick={jest.fn()}
			>
				Test
			</Button>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('should button to be "clickable"', () => {
		const mockFn = jest.fn();
		const wrapper = mount(
			<Button
				onClick={mockFn}
			>
				Test
			</Button>
		);

		wrapper.find('button').simulate('click');
		expect(mockFn).toHaveBeenCalled();
	});
});
