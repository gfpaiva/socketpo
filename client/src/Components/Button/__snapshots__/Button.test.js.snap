// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should mount properly 1`] = `
<Button>
  <button
    className="
			button
			
			
			
			
			"
  />
</Button>
`;

exports[`<Button /> should mount with children properly 1`] = `
<Button>
  <button
    className="
			button
			
			
			
			
			"
  >
    Test
  </button>
</Button>
`;

exports[`<Button /> should mount with props "className, medium, big, spaced, secondary, mx, onClick" 1`] = `
<Button
  big={true}
  className="test"
  medium={true}
  mx={true}
  onClick={[MockFunction]}
  secondary={true}
  spaced={true}
>
  <button
    className="test 
			button
			 button--big
			 button--medium
			 button--spaced
			 button--secondary
			 button--mx"
    onClick={[MockFunction]}
  >
    Test
  </button>
</Button>
`;
